Src/Particle/AMReX_ParticleMesh.H

```diff
@@ -69,18 +68,13 @@ ParticleToMesh (PC const& pc, MF& mf, int lev, F&& f)
                     f(pstruct[i], fabarr);
                 });
 
-                fab.atomicAdd<RunOn::Host>(local_fab, tile_box, tile_box, 0, 0, mf_pointer->nComp());
+                fab.atomicAdd<RunOn::Host>(local_fab, tile_box, tile_box, 0, 0, mf_tmp.nComp());
             }
         }
     }
 
-    mf_pointer->SumBoundary(pc.Geom(lev).periodicity());
-
-    if (mf_pointer != &mf)
-    {
-        mf.copy(*mf_pointer,0,0,mf_pointer->nComp());
-        delete mf_pointer;
-    }
+    mf_tmp.SumBoundary(pc.Geom(lev).periodicity());
+    mf.copy(mf_tmp,0,0,mf_tmp.nComp());
```

Comment:
Maybe we can do
```
mf.setVal(0.);
mf.ParallelAdd(mf_tmp, 0, 0, mf_tmp.nComp(), mf_tmp.nGrowVect(), IntVect(0), pc.Geom(lev).periodicity());
```
This will save one parallel communication if the grids of mf and mf_tmp are different.

Reply:
I am also confused now.  `mf.copy(mf_tmp,0,0,mf_tmp.nComp());` overrides what's in mf, whereas the title of the PR says `Dont zero out input multifab`.